crates/sui-types/src/messages.rs

```diff
@@ -931,6 +933,33 @@ pub struct BatchInfoRequest {
 #[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
 pub struct BatchInfoResponseItem(pub UpdateItem);
 
+/// Subscribe to notifications when new checkpoint certificates are available.
+///
+/// Note that there is no start field necessary, because checkpoint sequence numbers are
+/// contiguous. Therefore the client is always immediately sent the highest available checkpoint
+/// number, from which they can deduce if they are missing any checkpoints.
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct CheckpointInfoRequest {
+    // No request fields are currently necessary, but tonic errors when the request struct has size
+    // 0.
+    _ignored: u64,
```

Comment:
Can you describe a bit more on how we would switch from sync to follow for checkpoints? Curious why we are not specifying a starting checkpoint number here. What if the first checkpoint a node receives is already more than 1 checkpoint away from the local latest？

Reply:
There is no need for a starting number, because checkpoint sequences are contiguous. If you receive a summary with sequence N, you know that you need to sync all checkpoints less than N that you don't already have. Therefore there's never a need to receive anything other than the latest checkpoint.

The switch from sync to follow is coming in a subsequent PR, but the basic plan is:
- When sync starts, subscribe to the checkpoint stream, and begin syncing all missing checkpoints up to the latest known.
- Once all checkpoints have been synced, start the regular follower process, either from the beginning of the epoch, or from the correct most recent batch (as discussed on slack the other day).
- We continue monitoring the checkpoint stream, and continue syncing checkpoints as they come in. However, at this point, checkpoint sync will usually not require any new certificates, because we should have synced all the txes already via the follower.